torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -1243,14 +1255,12 @@ std::pair<std::vector<Val*>, bool> Index::getConsumerRootPredIndices(
   // If we are generating a predicate for initialization check if we should use
   // rfactor instead of root_dom
   bool use_rfactor = true;
-  if (consumer_tv->hasRFactor()) {
-    auto rfactor_dom = consumer_tv->getMaybeRFactorDomain();
+  if (consumer_tv->domain()->hasRFactor()) {
+    auto rfactor_dom = consumer_tv->domain()->rfactorDomain();
```

Comment:
I'd prefer to have `getMaybeRFactorDomain` in `kir::TensorView` as well.

Reply:
As I explained in the previous comments, I strongly feel we should avoid this. It may seem a bit convenient, but it ruins the class hierarchy.

Reply:
There's 28 uses of getMaybeRFactor. I appreciate your concerns, though I do really like it for convenience. It's a small but very common operation. I do believe consistency between the two levels of IR is more important than the convenience, so what ever the decision we should simply make it soon and work towards consistent interfaces no similar IR nodes.